_posts/2016-10-08-jumping-into-elasticsearch.markdown

```diff
@@ -0,0 +1,348 @@
+
+You can read more about these topics, as well as **shards** and **replicas**, in the [Getting Started](https://www.elastic.co/guide/en/elasticsearch/reference/current/_basic_concepts.html) guide.
+
+### Set up
+To simplify getting up and running, I'm going to assume you have [Docker](https://www.docker.com/) installed. (And if not... *why*?) With your daemon running, just throw this into the command line:
+
+```
+docker run -d -p 9200:9200 -p 5601:5601 psvet/elasticsearch-kibana-sense
+```
+
+This will result in an Elasticsearch cluster running on port `9200`, which you can verify with `curl <your docker host ip>:9200`. We also have [Kibana](https://www.elastic.co/products/kibana) available for viewing our data, as well as the [Sense](https://www.elastic.co/guide/en/sense/master/introduction.html) plugin for easily interacting with the REST API, running on `5601`.
+
+### In Action
+Now let's see a little of what it can do.
+
+In your browser, navigate to `<your docker host ip>:5601/app/sense`, and you'll see something like this:
+
+<div style="text-align:center;margin: 1rem auto;">
+  <img src="/assets/images/posts/jumping-into-elasticsearch/sense.png" alt="Sense" title="Sense" />
+</div>
+
+The [Elasticsearch REST API](https://www.elastic.co/guide/en/elasticsearch/reference/2.3/index.html) makes it possible to perform basic (and far-from-basic) CRUD operations at the index and document level, as well as monitor your clusters and nodes.
+
+### Creating an index
+First, let's create an index for a bare-bones social network. In the lefthand Sense panel paste (or type, and take advantage of the awesome IntelliSense):
+
+```
+PUT /socialnetwork
+{
+  "mappings": {
+    "user": {
+      "properties": {
+        "first_name": {
+          "type": "string"
+        },
+        "last_name": {
+          "type": "string"
+        },
+        "city": {
+          "type": "string"
+        },
+        "state": {
+          "type": "string"
+        },
+        "gender": {
+          "type": "string"
+        },
+        "age": {
+          "type": "integer"
+        },
+        "skills": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+```
+
+As you might as have guessed from the above JSON blob, in addition to creating the `socialnetwork` index, we explicitly mapped a `user` type. But I'll let you in on a secret: we could have left out the request object (`PUT /socialnetwork`), then simply created a `user` document and let Elasticsearch handle the mapping *dynamically*. We could have even started creating documents without initially creating the index! All we'd have to do is change the path to `/socialnetwork/user`, and provide the user data in the body straightaway.
```

Comment:
Reading as an outsider, this makes me think "That sounds easier, why didn't they just do it that way for the example?". It would be good to clarify that :)